src/sInv.sol

```diff
@@ -73,32 +96,43 @@ contract sINV is ERC4626 {
         address _inv,
         address _invMarket,
         address _gov,
-        uint _K
-    ) ERC4626(ERC20(_inv), "Staked Inv", "sasset") {
-        require(_K > 0, "_K must be positive");
+        address _guardian,
+        uint256 _K
+    ) ERC4626(ERC20(_inv), "Staked Inv", "sINV") {
+        if(_K == 0) revert KTooLow(_K, 1);
         IMarket(_invMarket).deposit(0); //creates an escrow on behalf of the sINV contract
-        invEscrow = IInvEscrow(IMarket(_invMarket).predictEscrow(address(this)));
+        invEscrow = IInvEscrow(IMarket(_invMarket).escrows(address(this)));
         invMarket = IMarket(_invMarket);
         DBR = ERC20(IMarket(_invMarket).dbr());
-        INV = ERC20(IMarket(_invMarket).collateral());
         gov = _gov;
-        targetK = _K;
+        guardian = _guardian;
+        kData.targetK = uint192(_K);
         prevK = _K;
         asset.approve(address(invMarket), type(uint).max);
```

Comment:
What's the purpose of this approval if we're sending INVs directly to the escrow instead of calling `market.deposit()`?

Reply:
Added reapprove function in [86d71b1](https://github.com/InverseFinance/sINV/pull/2/commits/86d71b14fff67e9d6a6aba7ecc61b5a96b2c4059)